helix-rest/src/main/java/org/apache/helix/rest/server/service/InstanceServiceImpl.java

```diff
@@ -234,8 +235,7 @@ private StoppableCheck performCustomInstanceCheck(String clusterId, String insta
 
   private Map<String, StoppableCheck> performPartitionsCheck(List<String> instances,
       RESTConfig restConfig, Map<String, String> customPayLoads) {
-    //TODO: move the heavy partition health preparation in separate class
-    PartitionHealth clusterPartitionsHealth = generatePartitionHealthMapFromZK();
+    PartitionHealth clusterPartitionsHealth = _dataAccessor.getPartitionHealthFromZK();
     // update the health status for those expired partitions on instances
     Map<String, List<String>> expiredPartitionsByInstance =
```

Comment:
I think we need to move this part as well. This is also part of read operation. We prefer to encapsulate the read logic (ZK read + direct API call) to one place.

Reply:
yeah, I thought about it as well. Ideally yes. The problem is I need to pass restconfig/baseurl/instances, a lot of params to the helix data accessor wrapper

Reply:
I dont think we need many. The baseurl comes from RESTConfig. So the only needed params are RESTConfig and list of instances.

Also you dont have to pass it in when construct the data accessor wrapper. You can make it as function inpit in getPartitionHealth(...).